Navigator: stabilize and export APIs

ciampo opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

Part of #59418
Supersedes #60927
Follow-up to #63317

Export the `Navigator` component, its subcomponents, and the `useNavigator` hook as stable APIs from the `@wordpress/components` package

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->


This is part of a [larger effort](https://github.com/WordPress/gutenberg/issues/59418) to remove `__experimental` prefix from all "experimental" components, effectively promoting them to regular stable components. See the related issue for more context.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

- Aside from the existing legacy exports, which use the `__experimental` prefix, add new, unprefixed exports:
  - Legacy exports stay untouched;
  - The new, stable exports will adopt slightly different names compared to legacy and the overloaded naming convention that we agreed to apply for component components:
    - `NavigatorProvider` => `Navigator`
    - `NavigatorScreen` => `Navigator.Screen`
    - `NavigatorButton` => `Navigator.Button`
    - `NavigatorBackButton` => `Navigator.BackButton`
    - `NavigatorToParentButton` won't be exported since it's already deprecated in the experimental version of the component
- Updated all JSDocs and README
- Updated Storybook and Unit Tests to use the stable version of the component
- Added Storybook redirect to avoid getting broken links
- A few smaller changes:
  - Updated the internal context namespacing (which shouldn't be a breaking change since context is an internal API of the `@wordpress/components` package)

## Next steps?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

- refactor existing usages in Gutenberg to stable version
- deprecate experimental version (both in JSDocs and with a runtime warning) (+ add unit tests for deprecation warnings)

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

- Technically there shouldn't be any runtime (behavioral) changes to `Navigator`;
- Make sure that Storybook examples are documented correctly and work as expected;
- Smoke test `Navigator` usages across the editor;
- Review JSDocs, READMEs, and IntelliSense snippets for both legacy and stable APIs

## ✍️  Dev note

The legacy set of `__experimentalNavigator*` APIs is deprecated and should instead be imported as `Navigator`. All of the sub-components are also available via the `Navigator` namespace.

Moreover, the `__experimentalNavigatorToParentButton` component and the `goToParent` method available via the `__experimentalUseNavigator` hook are now deprecated, and they now behave identically to the `__experimentalNavigatorBackButton` and the `goBack` method.

To recap:

- `__experimentalNavigatorProvider` => `Navigator`
- `__experimentalNavigatorScreen` => `Navigator.Screen`
-  `__experimentalNavigatorButton` => `Navigator.Button`
- `__experimentalNavigatorBackButton` => `Navigator.BackButton`
- `__experimentalNavigatorToParentButton` => `Navigator.BackButton`
- `__experimentalUseNavigator` => `useNavigator`

ciampo committed:
Remove experimental status from Storybook

ciampo committed:
NavigatorProvider => Navigator (internally)

ciampo committed:
Move legacy exports to separate file

ciampo committed:
Export overloaded notation

ciampo committed:
Use overloaded notation in Storybook

ciampo committed:
Use overloaded notation in unit tests (skipping deprecated component for now)

ciampo committed:
Update docs manifest.json

ciampo committed:
Navigator: update docs

ciampo committed:
Navigator.BackButton: update docs

ciampo committed:
NavigatorToParentButton: delete README

ciampo committed:
Navigator.Button: update docs

ciampo committed:
NavigatorScreen: update docs

ciampo committed:
Restore `NavigatorToParentButton` test

ciampo commented on packages/components/src/navigator/legacy.ts:
Legacy exports have been moved to a separate `legacy.ts` file (including the legacy JSDocs mentioning the components with their experimental export name).

This makes it easy to have separate names from the stable version, different JSDocs, and deprecation warnings.

ciampo commented on packages/components/src/navigator/index.ts:
New exports (and their up-to-date JSDocs) have been moved to the folder's main `index.ts` file

ciampo commented on packages/components/src/navigator/navigator-back-button/README.md:
All READMEs have also been updated:
- removed experimental snippet
- updated the naming of all `Navigator` sub-components, including internal links to other READMEs

ciampo commented on packages/components/src/navigator/navigator-provider/index.ts:
This file was actually renamed to `packages/components/src/navigator/navigator/index.tsx`, but GitHub is not picking that up

ciampo commented on packages/components/src/navigator/navigator/component.tsx:
```diff
@@ -273,41 +273,31 @@ function UnconnectedNavigatorProvider(
- * The `NavigatorProvider` component allows rendering nested views/panels/menus
- * (via the `NavigatorScreen` component and navigate between these different
- * view (via the `NavigatorButton` and `NavigatorBackButton` components or the
+ * The `Navigator` component allows rendering nested views/panels/menus
```
The top level `Navigator` component (previously named `NavigatorProvider`) is the only component for which I've kept a JSDoc for the internal implementation, because apparently Storybook uses it when displaying docs 🤷  Not sure why it doesn't use the one defined in `packages/components/src/navigator/index.ts`

ciampo replied:
it seems like the `ts` extension was the problem — changing it to `.tsx` seems to allow Storybook to pick the JSDoc up 😅  https://github.com/WordPress/gutenberg/pull/64613/commits/85471a9224994d8977221fc4915a159622c3f307


cc @mirka

ciampo commented on packages/components/src/navigator/stories/index.story.tsx:
```diff
@@ -10,20 +10,20 @@ import Button from '../../button';
+		// @ts-expect-error - See https://github.com/storybookjs/storybook/issues/23170
+		BackButton: Navigator.BackButton,
+	},
+	title: 'Components/Navigator',
```
Updated the `PREVIOUSLY_EXPERIMENTAL_COMPONENTS` in `storybook/manager-head.html` to make sure old links still work

ciampo committed:
Use dot notation in context namespaces too

ciampo committed:
Use named export for `NavigatorToParentButton` too

ciampo committed:
Fixup docs manifest

ciampo committed:
Export stable version from components package

ciampo committed:
CHANGELOG

ciampo committed:
Change index.ts extension to tsx to allow Storybook to extract JSDocs better, remove duplicate JSDocs

mirka commented on docs/manifest.json:
```diff
@@ -1103,22 +1103,16 @@
 		"markdown_source": "../packages/components/src/navigator/navigator-button/README.md",
 		"parent": "components"
 	},
-	{
-		"title": "NavigatorProvider",
-		"slug": "navigator-provider",
-		"markdown_source": "../packages/components/src/navigator/navigator-provider/README.md",
-		"parent": "components"
-	},
```
FYI, removing things from the manifest is not going to remove it from the [Block Editor Handbook](https://developer.wordpress.org/block-editor/reference-guides/components/) 😱 There are some extra steps, see https://github.com/WordPress/gutenberg/pull/60003#issuecomment-2010288927

ciampo replied:
Thank you! I'll follow-up accordingly

tyxla replied:
Just a reminder we'll need to take care of this one after merging.

ciampo replied:
Hey @WordPress/wordpress-org-meta-team , could you help with this? We need to update the Block Editor Handbook, removing the following components:

- [`NavigatorProvider`](https://developer.wordpress.org/block-editor/reference-guides/components/navigator-provider/)
- [`NavigatorButton`](https://developer.wordpress.org/block-editor/reference-guides/components/navigator-button/)
- [`NavigatorBackButton`](https://developer.wordpress.org/block-editor/reference-guides/components/navigator-back-button/)
- [`NavigatorScreen`](https://developer.wordpress.org/block-editor/reference-guides/components/navigator-screen/)
- [`NavigatorToParentButton`](https://developer.wordpress.org/block-editor/reference-guides/components/navigator-to-parent-button/)


If possible, we'd also like to set a redirect for those pages being removed. The redirect should point to [the `Navigator` component](https://developer.wordpress.org/block-editor/reference-guides/components/navigator/).


Could you please help with that? Thank you 🙏

adamwoodnz replied:
These updates have shipped. I noticed the formatting on the new doc loses consistency around the `Navigator.Button` point (heading is not bold):

![Screenshot 2024-10-01 at 10 42 06 AM](https://github.com/user-attachments/assets/37da3574-ee2c-4830-9ed8-fe50b6d62564)

ciampo replied:
Thank you!



> I noticed the formatting on the new doc loses consistency around the `Navigator.Button` point (heading is not bold):

Good catch! Opened https://github.com/WordPress/gutenberg/pull/65763 to fix it

mirka commented on packages/components/CHANGELOG.md:
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### New Features
```
Not "Enhancements"? (Enhancement == new feature)

ciampo replied:
My reasoning was "New API" (stable `Navigator`) => "New feature" section

Although I'm very much fine putting it under "Enhancements"

ciampo replied:
Done https://github.com/WordPress/gutenberg/pull/64613/commits/af97de9356dd7a8ba9a3ceeef705edebc0d36e71

mirka commented on packages/components/src/navigator/index.tsx:
````diff
@@ -0,0 +1,134 @@
+ * ```
+ */
+export const Navigator = Object.assign( TopLevelNavigator, {
+	dislayName: 'Navigator',
````
```suggestion
```

I can confirm this is unnecessary because it's misspelled 😄

ciampo replied:
Lol. That's one way to test changes out

ciampo replied:
Done https://github.com/WordPress/gutenberg/pull/64613/commits/81931212710d2112376156173907acaa10d7158a

mirka commented on packages/components/src/navigator/index.tsx:
````diff
@@ -0,0 +1,134 @@
+	 * );
+	 * ```
+	 */
+	Screen: Object.assign( NavigatorScreen, {
+		displayName: 'Navigator.Screen',
+	} ),
````
I actually like this better than doing `NavigatorScreen.displayName = 'Navigator.Screen'` in the individual subcomponent files. It makes more sense to handle that concern in this top-level index file. Should we update the contributing guide accordingly?

ciampo replied:
Sounds good!

ciampo replied:
https://github.com/WordPress/gutenberg/pull/64793

mirka commented on packages/components/src/navigator/index.tsx:
```diff
@@ -0,0 +1,134 @@
+ * view (via the `Navigator.Button` and `Navigator.BackButton` components or the
+ * `useNavigator` hook).
+ *
+ * @example
```
```suggestion
```

Should we remove this `@example` tag so the snippet shows up in the Storybook?

ciampo replied:
https://github.com/WordPress/gutenberg/pull/64613/commits/e7d20195ce125e78a2594874445bb34cb939766d

mirka commented on packages/components/src/navigator/navigator-back-button/component.tsx:
````diff
@@ -21,43 +21,7 @@ function UnconnectedNavigatorBackButton(
- * ```
- */
 export const NavigatorBackButton = contextConnect(
 	UnconnectedNavigatorBackButton,
-	'NavigatorBackButton'
+	'Navigator.BackButton'
 );
````
Hmm, this is an interesting one. If we do it this way, the context name is going to change for consumers of the legacy `NavigatorBackButton`, right? I wonder if we need separate context-connnected exports for the current and the legacy?

ciampo replied:
It is, yeah. although I didn't think it would be a problem — from the PR description:

> Updated the internal context namespacing (which shouldn't be a breaking change since context is an internal API of the `@wordpress/components` package)

The approach that you propose probably won't be enough, since we'd need to also change the `useContextSystem` call in the component render functions.

mirka replied:
Wow, you're right. I always assumed the context API was public! 😂

mirka commented on packages/components/src/navigator/test/index.tsx:
```diff
@@ -14,14 +14,14 @@ import { useState } from '@wordpress/element';
-	NavigatorScreen,
-	NavigatorButton,
-	NavigatorBackButton,
+	// NavigatorProvider,
+	// NavigatorScreen,
+	// NavigatorButton,
+	// NavigatorBackButton,
```
🧹 ?

ciampo replied:
🧹  https://github.com/WordPress/gutenberg/pull/64613/commits/0f2d077905f721ecb89023192f16656b4b08fccc

ciampo commented:
I'm going to resume work on this PR after https://github.com/WordPress/gutenberg/pull/64675 is merged.

tyxla commented:
> I'm going to resume work on this PR after #64675 is merged.

Sounds good. I was starting to review it, but I will halt it for now.

I wonder if it's a good idea to approach this modularly and split whatever we can to separate PRs to keep this one smaller.

ciampo committed:
Remove unnecessary and mispelled display name for top-level `Navigator`

ciampo committed:
Remove @example tag from top-level export

ciampo committed:
Clean up unused imports

ciampo commented:
I rebased and addressed simple feedback.

I would still wait before reviewing this PR. The current plan of action is:

- [ ] Review and merge https://github.com/WordPress/gutenberg/pull/64786
- [ ] Review and merge https://github.com/WordPress/gutenberg/pull/64777
- [ ] Rebase this PR and prepare for final review

I've also updated the [tracking issue](https://github.com/WordPress/gutenberg/issues/59418) adding all of the planned follow-up step to this PR.

ciampo committed:
Fix storybook styles

ciampo committed:
Update new section of the docs

ciampo committed:
Update more docs

ciampo committed:
Update deprecation warning

ciampo commented:
Rebased on top of trunk to include the latest changes.



> > I wonder if it's a good idea to approach this modularly and split whatever we can to separate PRs to keep this one smaller.
> 
> @tyxla , Even if the size of the PR is medium-large, I believe 99% of the changes left in this PR are simple renames — and in that sense I feel ok with the current size. Having said that, I'll re-assess once all dependency PRs are merged, before asking for a final round of review.

I split some code changes to #65671, which is ready for review.

ciampo committed:
Update and align docs

ciampo committed:
Fix tests warning checks

ciampo committed:
Typo

ciampo commented:
Rebased to include changes from https://github.com/WordPress/gutenberg/pull/65671

This PR is ready for a final round of review @tyxla @mirka

tyxla commented on packages/components/CHANGELOG.md:
```diff
@@ -17,6 +17,7 @@
 -   `MenuGroup`: Simplify the MenuGroup styles within dropdown menus. ([#65561](https://github.com/WordPress/gutenberg/pull/65561)).
 -   `DatePicker`: Use compact button size. ([#65653](https://github.com/WordPress/gutenberg/pull/65653)).
 -   `Navigator`: add support for exit animation ([#64777](https://github.com/WordPress/gutenberg/pull/64777)).
+-   `Navigator`: stabilize APIs ([#64613](https://github.com/WordPress/gutenberg/pull/64613)).
```
I think it makes sense to list it under "New features" as well, now that it's stable

ciampo replied:
We had already [discussed](https://github.com/WordPress/gutenberg/pull/64613#discussion_r1726677282) which CHANGELOG section to put this under, and we went for Enhancements (also, similar PRs also added an entry under Enhancements)

tyxla replied:
I just think it makes sense since it's part of the formalization of the stable API, but I have no strong feelings about it. Leaving to your judgment.

ciampo replied:
I will favour consistency and leave it under the `Enhancements` section, as we've done for other PRs

tyxla commented on packages/components/src/navigator/index.tsx:
```diff
@@ -0,0 +1,132 @@
+	/**
+	 * The `Navigator.Button` component can be used to navigate to a screen and
+	 * should be used in combination with the `Navigator`, the `Navigator.Screen`
+	 * and the `Navigator.BackButton` components, and the `useNavigator` hook.
```
We claim that these components should be used also with `useNavigator`, but then should we include the hook usage in the examples?

ciampo replied:
Good point. TBH, `useNavigator` is more of an advanced usage (and there's an example of it in Storybook). Do you think that's enough, or should I look into making the snippet longer and incorporate `useNavigator`?

tyxla replied:
Or we can just exclude the hook from the description?

ciampo replied:
Sure, I can do that

ciampo replied:
Mentions removed in https://github.com/WordPress/gutenberg/pull/64613/commits/3c3df877bd654c92a8898d363c047e06934dc276.

Folks can always refer to [this section](https://github.com/WordPress/gutenberg/blob/3c3df877bd654c92a8898d363c047e06934dc276/packages/components/src/navigator/navigator/README.md#the-navigator-object) in the main README for docs about the `useNavigator` hook (plus, I've added better JSDocs in https://github.com/WordPress/gutenberg/pull/64613/commits/567f41879cd72147f8db95be614c7fafeddf2d6b)

tyxla commented on docs/manifest.json:
```diff
@@ -1109,22 +1109,16 @@
 		"markdown_source": "../packages/components/src/navigator/navigator-button/README.md",
```
I see we continue calling these for example `NavigatorButton`, but they're actually exposed as `Navigator.Button`? Should this be reflected in the manifest? Should we just leave the single `Navigator` and have all READMEs in it?

ciampo replied:
I think the name is auto-generated from the filename or something else. I like the idea of moving everything under one README, though. We currently have some compound components having one unique README, and other compound components with one README per sub-component. It would be good if we agreed on an approach going forward — @WordPress/gutenberg-components , any preferences?

ciampo replied:
I merged all READMEs into one: https://github.com/WordPress/gutenberg/pull/64613/commits/5e33f26ba86f2308b331a0b7af60feb931836727, https://github.com/WordPress/gutenberg/pull/64613/commits/0cff1065813626fde5a2e14f444e69dbe39c6bfd and https://github.com/WordPress/gutenberg/pull/64613/commits/cea2b8e66c939f20fbdaa931ec501c7991622cf7.

I think I like keeping one README more, but happy to hear other's thoughts before merging the PR.

mirka replied:
> I think I like keeping one README more

No strong opinion. I might have an updated opinion after I attempt auto-generating the READMEs 🙈

tyxla replied:
I prefer a single README too - primarily because it's a better developer experience. It's way easier to understand the entire component family, rather than having to jump between multiple READMEs and figure out the puzzle on your own.

ciampo replied:
Sounds good — I guess we can go with the one README for now and tweak later if necessary, together with the rest of the components that will need similar changes.

tyxla commented on packages/components/src/index.ts:
```diff
@@ -132,6 +132,7 @@ export {
 	NavigatorToParentButton as __experimentalNavigatorToParentButton,
 	useNavigator as __experimentalUseNavigator,
```
Are we planning to deprecate the legacy ones at some point? Perhaps in a separate PR after we update the existing usages?

ciampo replied:
Yup! From the "Next steps" section in the PR description:

> - refactor existing usages in Gutenberg to stable version
> - deprecate experimental version (both in JSDocs and with a runtime warning) (+ add unit tests for deprecation warnings)

tyxla replied:
Thanks for confirming 😉 It's a long-going PR, so it never hurts to confirm.

tyxla commented on packages/components/src/navigator/navigator-back-button/component.tsx:
```diff
@@ -23,5 +23,5 @@ function UnconnectedNavigatorBackButton(
 export const NavigatorBackButton = contextConnect(
 	UnconnectedNavigatorBackButton,
-	'NavigatorBackButton'
+	'Navigator.BackButton'
```
Double-checking if this is OK - I found it unexpected that here we use the `Navigator.BackButton` context name, but in the button hook, we use `NavigatorBackButton` without the dot notation:

https://github.com/WordPress/gutenberg/blob/fdff00c9b442dd4263da62a7dd8a3f9fe617b3ad/packages/components/src/navigator/navigator-back-button/hook.ts#L23

Same for the rest of the subcomponents and their context.

ciampo replied:
Nope, that's a bug — well spotted! I'm going to push an update

ciampo replied:
Fixed in https://github.com/WordPress/gutenberg/pull/64613/commits/d153bcf0db769fb57c960c90301e09a2ba24c6c2

tyxla reviewed (commented):
This looks good and tests well to me 👍 Great work @ciampo 🚀 

I've left a few questions before shipping, most are just for confirmation of my assumptions.

ciampo committed:
Fix useContextSystem names

ciampo committed:
Remove mentions of the `useNavigator` hook

ciampo committed:
useNavigator JSDocs

ciampo committed:
Move all READMEs under the main README

ciampo committed:
Move README to component's root folder

ciampo committed:
Update types JSDocs to match README

ciampo committed:
Update docs manifest

ciampo commented:
All feedback should be addressed, including [this conversation](https://github.com/WordPress/gutenberg/pull/64613#discussion_r1777390024) on keeping one vs many READMEs.

tyxla commented on packages/components/src/navigator/README.md:
```diff
@@ -0,0 +1,168 @@
+
+###### `path`: `string`
+
+The screen&quot;s path, matched against the current path stored in the navigator.
```
```suggestion
The screen's path, matched against the current path stored in the navigator.
```

tyxla commented on packages/components/src/navigator/README.md:
```diff
@@ -0,0 +1,168 @@
+# `Navigator`
+
+`Navigator` is a collection components that allow rendering nested views/panels/menus (via the [`Navigator.Screen` component](/packages/components/src/navigator/navigator-screen/README.md)) and navigate between them (via the [`Navigator.Button`](/packages/components/src/navigator/navigator-button/README.md) and [`Navigator.BackButton`](/packages/components/src/navigator/navigator-back-button/READ ...
```
Didn't we remove these READMEs? 

```suggestion
`Navigator` is a collection of components that allow rendering nested views/panels/menus (via the `Navigator.Screen`) and navigate between them (via the `Navigator.Button` and `Navigator.BackButton` components).
```

ciampo replied:
🤦

tyxla commented on packages/components/src/navigator/README.md:
```diff
@@ -0,0 +1,168 @@
+
+The `goBack` function allows navigating to the parent screen. Parent/child navigation only works if the paths you define are hierarchical (see note above).
+
+When a match is not found, the function will try to recursively navigate the path hierarchy until a matching screen (or the root `/`) are found.
```
```suggestion
When a match is not found, the function will try to recursively navigate the path hierarchy until a matching screen (or the root `/`) is found.
```

tyxla commented on packages/components/src/navigator/README.md:
```diff
@@ -0,0 +1,168 @@
+
+###### `location`: `NavigatorLocation`
+
+The `location` object represent the current location, and has a few properties:
```
```suggestion
The `location` object represents the current location, and has a few properties:
```

tyxla commented on packages/components/src/navigator/README.md:
```diff
@@ -0,0 +1,168 @@
+
+`Navigator` assumes that screens are organized hierarchically according to their `path`, which should follow a URL-like scheme where each path segment starts with and is separated by the `/` character.
+
+`Navigator` will treat "back" navigations as going to the parent screen — it is therefore responsibility of the consumer of the component to create the correct screen hierarchy.
```
```suggestion
`Navigator` will treat "back" navigations as going to the parent screen — it is, therefore, the responsibility of the consumer of the component to create the correct screen hierarchy.
```

tyxla commented on packages/components/src/navigator/README.md:
```diff
@@ -0,0 +1,168 @@
+
+`Navigator` assumes that screens are organized hierarchically according to their `path`, which should follow a URL-like scheme where each path segment starts with and is separated by the `/` character.
+
+`Navigator` will treat "back" navigations as going to the parent screen — it is therefore responsibility of the consumer of the component to create the correct screen hierarchy.
```
```suggestion
`Navigator` will treat "back" navigations as going to the parent screen — it is, therefore, the responsibility of the consumer of the component to create the correct screen hierarchy.
```

tyxla commented on packages/components/src/navigator/test/index.tsx:
```diff
@@ -286,30 +286,30 @@ const MyNavigation = ( {
-				</NavigatorScreen>
+				</Navigator.Screen>
 
 				{ /* A `NavigatorScreen` with `path={ PATHS.NOT_FOUND }` is purposefully not included. */ }
```
```suggestion
				{ /* A `Navigator.Screen` with `path={ PATHS.NOT_FOUND }` is purposefully not included. */ }
```

tyxla commented on packages/components/src/navigator/stories/index.story.tsx:
```diff
@@ -40,7 +40,7 @@ const meta: Meta< typeof NavigatorProvider > = {
+						[data-wp-component="Navigator"] {
 							height: 250px;
 						}
 						[data-wp-component="NavigatorScreen"]:not([data-sticky]) {
```
I think we need to change this one too:

```suggestion
						[data-wp-component="Navigator.Screen"]:not([data-sticky]) {
```

ciampo replied:
The current version is correct — we're setting the height on the wrapper, as also suggested in the README. Each screen, if the wrapper's height is set, will follow that height thanks to the grid layout.

tyxla replied:
I think you're looking at the wrong place - this is about the padding, and in my testing the padding appears only if we use the dot notation.

tyxla replied:
Alternatively, feel free to remove this style if you think we don't need it. But my point was that if you want the padding, you need to add the dot.

ciampo replied:
You're right — I looked at the wrong diff 😅 

I pushed a fix as you suggested.

tyxla commented on packages/components/src/navigator/types.ts:
```diff
@@ -100,6 +100,24 @@ export type NavigatorProps = {
+	 * to their `path`, which should follow a URL-like scheme where each path
+	 * segment starts with and is separated by the `/` character.
+	 *
+	 * `Navigator` will treat "back" navigations as going to the parent screen —
+	 * it is therefore responsibility of the consumer of the component to create
+	 * the correct screen hierarchy.
```
```suggestion
	 * `Navigator` will treat "back" navigations as going to the parent screen —
	 * it is, therefore, the responsibility of the consumer of the component to
	 * create the correct screen hierarchy.
```

tyxla reviewed (approved):
Looking great, excited to finally stabilize the `Navigator` family 👍 

Left a few minor suggestions, mostly grammar, typos, etc.

Thank you 🚀

ciampo committed:
Fix JSDocs typos, grammar, and broken links.

ciampo committed:
Update storybook selector
